### PR TITLE
Fixes #4759

### LINKED ISSUE #4759: Reaction parser fails when CX extensions are present
**Describe the bug**
```
In [7]: rxn = rdChemReactions.ReactionFromSmarts('[C:1]Br.[C:2]O>>[C:2][C:1] |$Aryl;;;;;Aryl$|')
[17:53:35] 

****
Range Error
idx
Violation occurred on line 206 in file /home/conda/feedstock_root/build_artifacts/rdkit_1635252119815/work/Code/GraphMol/ROMol.cpp
Failed Expression: 5 < 2
****

```

**Expected behavior**
We should eventually be able to parse that information, but until we can it should be ignored.

**Configuration (please complete the following information):**
 - RDKit version: 2021.09
 - OS: all

### PR DESCRIPTION
**Pull request by greglandrum**
Minor change to strip out extra stuff at the end of a reaction SMARTS/SMILES since we can't parse it anyway

### CODE REVIEW

**Review by ricrogz — APPROVED** [review]
LGTM

File: Code/GraphMol/ChemReactions/DaylightParser.cpp [hunk 1]
```diff
@@ -141,6 +152,19 @@ ChemicalReaction *RxnSmartsToChemicalReaction(
   std::vector<std::string> productSmarts =
       DaylightParserUtils::splitSmartsIntoComponents(productText);
 
+  // if the input includes CX extensions, they will show up here after the
+  // last product. We can't currently deal with those, so issue a warning and
+  // strip out anything following the last productSmarts
+  // This was Github #4759
```
Comment (bp-kelley) [inline Code/GraphMol/ChemReactions/DaylightParser.cpp:158]: The comment indicates a warning is issued but the code doesn’t issue a warning.

**Review by bp-kelley — CHANGES_REQUESTED** [review]

File: Code/GraphMol/ChemReactions/DaylightParser.cpp (hunk 1 above)
Comment (greglandrum) [reply Code/GraphMol/ChemReactions/DaylightParser.cpp:158]: doh!
Thanks for noticing that.

Comment (greglandrum) [reply Code/GraphMol/ChemReactions/DaylightParser.cpp:158]: @bp-kelley : that's now fixed

File: Code/GraphMol/ChemReactions/DaylightParser.cpp [hunk 2]
```diff
@@ -141,6 +152,22 @@ ChemicalReaction *RxnSmartsToChemicalReaction(
...
+  if (!productSmarts.empty()) {
+    auto &lastSmarts = productSmarts.back();
+    boost::trim(lastSmarts);
+    std::vector<std::string> tokens;
+    boost::split(tokens, lastSmarts, boost::is_any_of(" \t"));
+    if (tokens.size() > 1) {
+      BOOST_LOG(rdWarningLog)
+          << "stripping extra text from input data" << std::endl;
```
Comment (bp-kelley) [inline Code/GraphMol/ChemReactions/DaylightParser.cpp:166]: Maybe smarts reaction as opposed to input data?  Might be more useful in a log.

**Review by bp-kelley — CHANGES_REQUESTED** [review]

**Review by bp-kelley — APPROVED** [review]

**Pull request merged**